kmath-tensors/src/commonMain/kotlin/space/kscience/kmath/tensors/api/StatisticTensorAlgebra.kt

```diff
@@ -0,0 +1,137 @@
+     *
+     * @param dim the dimension to reduce.
+     * @param keepDim whether the output tensor has [dim] retained or not.
+     * @return the minimum value of each row of the input tensor in the given dimension [dim].
+     */
+    public fun Tensor<T>.min(dim: Int, keepDim: Boolean): DoubleTensor
+
+    /**
+     * Returns the maximum value of all elements in the input tensor.
+     *
+     * @return the maximum value of all elements in the input tensor.
+     */
+    public fun Tensor<T>.max(): Double
+
+    /**
+     * Returns the maximum value of each row of the input tensor in the given dimension [dim].
+     *
+     * If [keepDim] is true, the output tensor is of the same size as
+     * input except in the dimension [dim] where it is of size 1.
+     * Otherwise, [dim] is squeezed, resulting in the output tensor having 1 fewer dimension.
+     *
+     * @param dim the dimension to reduce.
+     * @param keepDim whether the output tensor has [dim] retained or not.
+     * @return the maximum value of each row of the input tensor in the given dimension [dim].
+     */
+    public fun Tensor<T>.max(dim: Int, keepDim: Boolean): DoubleTensor
+
+    /**
+     * Returns the sum of all elements in the input tensor.
+     *
+     * @return the sum of all elements in the input tensor.
+     */
+    public fun Tensor<T>.sum(): Double
+
+    /**
+     * Returns the sum of each row of the input tensor in the given dimension [dim].
+     *
+     * If [keepDim] is true, the output tensor is of the same size as
+     * input except in the dimension [dim] where it is of size 1.
+     * Otherwise, [dim] is squeezed, resulting in the output tensor having 1 fewer dimension.
+     *
+     * @param dim the dimension to reduce.
+     * @param keepDim whether the output tensor has [dim] retained or not.
+     * @return the sum of each row of the input tensor in the given dimension [dim].
+     */
+    public fun Tensor<T>.sum(dim: Int, keepDim: Boolean): DoubleTensor
+
+    /**
+     * Returns the mean of all elements in the input tensor.
+     *
+     * @return the mean of all elements in the input tensor.
+     */
+    public fun Tensor<T>.mean(): Double
+
+    /**
+     * Returns the mean of each row of the input tensor in the given dimension [dim].
+     *
+     * If [keepDim] is true, the output tensor is of the same size as
+     * input except in the dimension [dim] where it is of size 1.
+     * Otherwise, [dim] is squeezed, resulting in the output tensor having 1 fewer dimension.
+     *
+     * @param dim the dimension to reduce.
+     * @param keepDim whether the output tensor has [dim] retained or not.
+     * @return the mean of each row of the input tensor in the given dimension [dim].
+     */
+    public fun Tensor<T>.mean(dim: Int, keepDim: Boolean): DoubleTensor
+
+    /**
+     * Returns the standard deviation of all elements in the input tensor.
+     *
+     * @return the standard deviation of all elements in the input tensor.
```

Comment:
```suggestion
     * Returns the standard deviation of all elements in the input tensor.
```

This is not necessary.